hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/onset/WorkflowOnset.java

```diff
@@ -15,16 +15,140 @@
+ * ByteBuffer} and checks it.
+ */
 public class WorkflowOnset {
 
+    private final OnsetChecker checker;
+
+    /**
+     * Constructor of {@code WorkflowOnset}
+     *
+     * @param checker the {@link OnsetChecker}
+     * @throws NullPointerException if one of the arguments is {@code null}
+     */
+    public WorkflowOnset(@NonNull final OnsetChecker checker) {
+        this.checker = requireNonNull(checker);
+    }
+
+    /**
+     * Parse the given {@link ByteBuffer} and check its validity
+     *
+     * @param ctx the {@link SessionContext}
+     * @param buffer the {@code ByteBuffer} with the serialized transaction
+     * @return an {@link OnsetResult} with the parsed and checked entities
+     * @throws PreCheckException if the data is not valid
+     * @throws NullPointerException if one of the arguments is {@code null}
+     */
+    public OnsetResult parseAndCheck(
+            @NonNull final SessionContext ctx, @NonNull final ByteBuffer buffer)
+            throws PreCheckException {
+        requireNonNull(ctx);
+        requireNonNull(buffer);
+
+        return doParseAndCheck(ctx, () -> ctx.txParser().parseFrom(buffer));
+    }
+
+    /**
+     * Parse the given {@link ByteBuffer} and check its validity
+     *
+     * @param ctx the {@link SessionContext}
+     * @param buffer the {@code ByteBuffer} with the serialized transaction
+     * @return an {@link OnsetResult} with the parsed and checked entities
+     * @throws PreCheckException if the data is not valid
+     * @throws NullPointerException if one of the arguments is {@code null}
+     */
     public OnsetResult parseAndCheck(
             @NonNull final SessionContext ctx, @NonNull final byte[] buffer)
             throws PreCheckException {
-        throw new UnsupportedOperationException("Not implemented");
+        requireNonNull(ctx);
+        requireNonNull(buffer);
+
+        return doParseAndCheck(ctx, () -> ctx.txParser().parseFrom(buffer));
+    }
+
+    @SuppressWarnings("deprecation")
+    private OnsetResult doParseAndCheck(
+            @NonNull final SessionContext ctx, @NonNull final TransactionSupplier txSupplier)
+            throws PreCheckException {
+
+        // 1. Parse the transaction object
+        final Transaction tx;
+        try {
+            tx = txSupplier.parse();
+        } catch (InvalidProtocolBufferException e) {
+            throw new PreCheckException(INVALID_TRANSACTION);
+        }
+
+        checker.checkTransaction(tx);
+
+        // 2. Parse and validate the signed transaction (if available)
+        final ByteString bodyBytes;
+        final SignatureMap signatureMap;
+        if (!tx.getSignedTransactionBytes().isEmpty()) {
+            final SignedTransaction signedTransaction =
+                    parse(ctx.signedParser(), tx.getSignedTransactionBytes(), INVALID_TRANSACTION);
+            checker.checkSignedTransaction(signedTransaction);
```

Comment:
I'll need to verify with @tinker-michaelj or @Neeharika-Sompalli, but the new pre-handlers may also be checking this signature. Assuming that's true, should this signature be checked here, there, or in both places?